CMakeLists.txt

```diff
@@ -147,38 +117,62 @@ SET(CMAKE_C_STANDARD 11)
+# PIE compilation (Position Independent Executables) results in the
+# `file` utility incorrectly recognising the compiled vegastrike-engine binary
+# as a shared library instead of a position independent shared executable
+# object.
+#
+# The effect of the bug is that vegastrike-engine can still be started from the
+# command line but that it will not be recognised as an executable by GUI
+# file managers such as Nautilus and Dolphin.
+#
+# To avoid this scenario, turn off this flag by default and let packagers
+# on other distributions turn this on if their OS is able to correctly deal
+# with Position Independent Executables.
+
+# For more info, see:
+# - https://bugs.launchpad.net/ubuntu/+source/file/+bug/1747711
+# - https://github.com/vegastrike/Vega-Strike-Engine-Source/issues/94
+#
+OPTION(ENABLE_PIE "Enable Position Independent Executables/Shared Libraries (NOT RECOMMENDED on Ubuntu/Mint)" OFF)
+MESSAGE("-- Always using preferred PIE logic now")
+
+include(CheckPIESupported)
+check_pie_supported()
+UNSET(CMAKE_POSITION_INDEPENDENT_CODE)
+IF (ENABLE_PIE)
+    MESSAGE("!! Enabling Position Independent Executables/Shared Libraries (NOT RECOMMENDED on Ubuntu/Mint) !!")
+    SET(CMAKE_POSITION_INDEPENDENT_CODE ON)
+ELSE (ENABLE_PIE)
+    MESSAGE("++ Disabling Position Independent Executables/Shared Libraries (Recommended on Ubuntu/Mint)")
+    SET(CMAKE_POSITION_INDEPENDENT_CODE OFF)
+ENDIF (ENABLE_PIE)
+
 # Should we install gtest?
 OPTION(INSTALL_GTEST "Should we download and install GTest?" ON)
 
 # Should we run gtest?
 OPTION(USE_GTEST "Should we build and run the unit tests using GTest?" ON)
 
 IF (INSTALL_GTEST OR USE_GTEST)
-    MESSAGE("-- Configuring Unit Tests")
+    MESSAGE(STATUS "Configuring Unit Tests")
     ENABLE_TESTING()
 
     IF (INSTALL_GTEST)
         INCLUDE(FetchContent)
         FetchContent_Declare(
             googletest
             GIT_REPOSITORY https://github.com/google/googletest.git
-            GIT_TAG        release-1.11.0
+            GIT_TAG        v1.15.2
         )
 
-        FetchContent_GetProperties(googletest)
-        STRING(TOLOWER "googletest" lcName)
-        IF(NOT ${lcName}_POPULATED)
-            # Fetch the content using previously declared details
-            FetchContent_Populate(googletest)
-
-            # Set custom variables, policies, etc.
+        # Set custom variables, policies, etc.
 
-            # For Windows: Prevent overriding the parent project's compiler/linker settings
-            SET(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+        # For Windows: Prevent overriding the parent project's compiler/linker settings
+        SET(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
 
-            # Bring the populated content into the build
-            ADD_SUBDIRECTORY(${${lcName}_SOURCE_DIR} ${${lcName}_BINARY_DIR})
-        ENDIF(NOT ${lcName}_POPULATED)
+        FetchContent_MakeAvailable(googletest)
```

Comment:
nice that this simplifies down too

Reply:
Agreed

Reply:
That was an old CMake construct needed for CMake backwards compatibility. Now that CMake v3.21 is our minimum, we can use the newer, better way of doing it.